test_naucse_render/test_load.py

```diff
@@ -73,3 +74,38 @@ def test_isolated_load(tmp_path):
 
     data = read_yaml(tmp_path, 'test.yaml')
     assert data == {'data': {'a': 1, 'b': 2}}
+
+
+def test_read_yaml_disallow_duplicate_keys(tmp_path):
+    """Assert that read_yaml disallows duplicate keys"""
+    yaml_path = tmp_path / 'test.yaml'
+    yaml_path.write_text("""data:
+        a: 1
+        a: 2
+    """)
+    with pytest.raises(yaml.constructor.ConstructorError):
+        read_yaml(tmp_path, 'test.yaml')
+
+
+@pytest.mark.xfail(
+    strict=True,
+    reason="Incomplete workaround for https://github.com/yaml/pyyaml/issues/165"
+)
+def test_read_yaml_allow_merge(tmp_path):
+    """Assert that read_yaml allows overriding dict keys in YAML merge"""
+    yaml_path = tmp_path / 'test.yaml'
+    yaml_path.write_text("""data:
+        1:
+            <<: &common
```

Comment:
Use YAML, they said. It is easy -- like plaintext, they said.

Reply:
Well, YAML is the Ruby of markup languages :)
But this particular feature doesn't bother me, actually.